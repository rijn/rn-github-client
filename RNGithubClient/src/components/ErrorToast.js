import { Toast } from 'native-base';

export default (e) => {
  Toast.show({
    text: e.message,
    position: 'top',
    buttonText: 'Okay',
    type: 'danger',
    duration: 3000
  });
}