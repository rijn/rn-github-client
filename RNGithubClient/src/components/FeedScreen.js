import React from 'react';
import { connect } from 'react-redux';
import { StyleSheet, View } from 'react-native';
import { Button, Text } from 'native-base';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
  },
});

/**
 * Class for feed screen.
 *
 * @desc All user activities
 *
 * @class FeedScreen
 */
class FeedScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    let { dispatch } = this.props;

    return (
      <View style={styles.container}>
        <Text>Main</Text>
      </View>
    );
  }
};

export default connect()(FeedScreen);