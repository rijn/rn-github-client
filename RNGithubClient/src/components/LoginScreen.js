import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, AsyncStorage } from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import { Container, Header, Body, Title, Content, Form, Label, Item, Button, Input, Text } from 'native-base';

const styles = StyleSheet.create({
  logoContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 100
  },
  form: {
    paddingBottom: 10
  },
  button: {
    margin: 10
  }
});

const LoginScreen = class extends React.Component {
  constructor(props) {
    super(props);

    this.navigation = props.navigation;

    this.checkLocalStorage();

    this.state = {
      user: null
    };
  }

  login() {
    this.navigation.dispatch({ type: 'Login', params: { user: this.state.user } });
  }

  checkLocalStorage() {
    AsyncStorage.getItem('@githubClient:user').then((user) => {
      if (user){
        this.setState({ user });
        this.login();
      }
    });
  }

  render() {
    return (
      <Container>
        <Header>
          <Body>
            <Title>Login</Title>
          </Body>
        </Header>
        <Content>
          <View style={styles.logoContainer}>
            <Entypo name='github' size={100} />
          </View>
          <Form style={styles.form}>
            <Item floatingLabel last>
              <Label>Username</Label>
              <Input onChangeText={(t) => this.setState({ user: t })}/>
            </Item>
          </Form>
          <Button block style={styles.button}
            onPress={() => this.login()}>
            <Text>GO</Text>
          </Button>
        </Content>
      </Container>
    );
  }
};

LoginScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

export default LoginScreen;