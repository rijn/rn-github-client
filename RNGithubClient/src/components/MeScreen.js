import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View } from 'react-native';
import Entypo from 'react-native-vector-icons/Entypo';
import { Container, Content, Separator, ListItem, Text, Left, Body, Right, Thumbnail } from 'native-base';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff'
  }
});

class MeScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    if (!this.props.userInfo) return (<View></View>);

    let { userInfo: { error, loading }, dispatch } = this.props;

    if (error) {
      return <Text>{error.toString()}</Text>;
    } else if (loading) {
      return <Text>Loading</Text>;
    } else {
      let { userInfo: { user: { login, avatarUrl, name, bio, company } } } = this.props;

      return (
        <Container>
          <Content>
            <View style={styles.container}>
              <Separator bordered>
                <Text>Profile</Text>
              </Separator>
              <ListItem avatar>
                <Left>
                  <Thumbnail source={{ uri: avatarUrl }} />
                </Left>
                <Body>
                  <Text>{name}</Text>
                  <Text note>{bio}</Text>
                  <Text note>{company}</Text>
                </Body>
                <Right>
                </Right>
              </ListItem>
              <ListItem last onPress={() => dispatch({ type: 'UserProfile', params: { user: { login } } })}>
                <Text>View profile</Text>
              </ListItem>
            </View>
          </Content>
        </Container>
      );
    }
  }
};

MeScreen.propTypes = {
  userInfo: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    user: PropTypes.object,
  }).isRequired
};

const GetUserInfoQuery = gql`
  query getUserInfo($login: String!) {
    user(login: $login) {
      login
      name
      avatarUrl
      bio
      company
      email
    }
  }
`;

const withInfo = graphql(GetUserInfoQuery, {
  name: 'userInfo',
  skip: (ownProps) => !ownProps.isLoggedIn,
  options: (props) => ({
    pollInterval: 20000,
    variables: {
      login: props.user
    }
  })
});

MeScreen = withInfo(MeScreen);

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  user: state.auth.user,
});

export default connect(mapStateToProps)(MeScreen);