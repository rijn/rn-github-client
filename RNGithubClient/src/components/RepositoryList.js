import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { StyleSheet, View, FlatList, Dimensions } from 'react-native';
import {
  Container, Content, Separator, List, ListItem, Text, Left, Body, Right,
  Thumbnail, Button, Spinner
} from 'native-base';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

const { height } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    height: height - 64 - 50
  },
  inline: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center'
  },
  ceil: {
    marginRight: 10
  }
});

/**
 * List of repositories screen
 *
 * @class RepositoryList
 */
class RepositoryList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      composed: null,
      refreshing: true,
      edges: []
    };
  }

  /**
   * Receive new props and calculate and concat the list
   *
   * @param  {Props} newProps
   */
  componentWillReceiveProps(newProps) {
    // this.setState({ refreshing: !!newProps.loading });
    // if (newProps.loading) return;

    let composed = newProps.repos ? newProps.repos : newProps.starredRepos ? newProps.starredRepos : null;
    let edges = _.get(composed, 'repository.edges');
    if (edges) {
      this.setState({ composed, edges });
    }
    if (this.state.refreshing == true && !!composed) {
      this.setState({ refreshing: false });
    }

    this.props = newProps;
  }

  /**
   * Primary language of repo component
   *
   * @param  {Object} primaryLanguage
   * @return {ReactDOM}
   */
  primaryLanguageView(primaryLanguage) {
    if (!primaryLanguage) return;
    let { name: primaryLanguageName, color: primaryLanguageColor } = primaryLanguage;
    return (
      <View style={[ styles.inline, styles.ceil ]}>
        <View style={{
          width: 7, height: 7, borderRadius: 7,
          marginRight: 5,
          backgroundColor: primaryLanguageColor
        }}></View>
        <Text note>{primaryLanguageName}</Text>
      </View>
    );
  }

  handleRefresh() {
    this.setState({ edges: [], refreshing: true });
    this.state.composed.refetch();
  }

  /**
   * Render function
   *
   * @return {ReactDOM}
   */
  render() {
    let { dispatch } = this.props;

    let { composed, edges, loading, refreshing } = this.state;

    let repositoryList = _.chain(edges).map(({ node }) => node).value();

    return (
      <List scrollEnabled={false} style={styles.container}>
        <FlatList
          data={repositoryList}
          onEndReached={() => { if (!this.props.loading) { return composed.fetchNextPage(); } }}
          keyExtractor={({ id }) => id}
          refreshing={refreshing}
          onRefresh={() => { this.handleRefresh(); }}
          renderItem={({
            item: {
              id, name, description, isFork,
              parent,
              primaryLanguage,
              updatedAt,
              owner: { login: owner },
              stargazers: { totalCount: stars },
              forks: { totalCount: forks }
            }
          }) => (
            <ListItem onPress={() => dispatch({ type: 'RepositoryProfile', params: { repository: { name, owner } } })}>
              <View>
                <View style={styles.inline}>
                  <Text>{name}</Text>
                  {isFork && <Text note>{parent.nameWithOwner}</Text>}
                </View>
                {description && <View style={[ styles.inline, { marginTop: 5 } ]}>
                  <Text note>{description}</Text>
                </View>}
                <View style={[ styles.inline, { marginTop: 5 } ]}>
                  {this.primaryLanguageView(primaryLanguage)}
                  <View style={[ styles.inline, styles.ceil ]}>
                    <Ionicons name='ios-star-outline' style={{ marginRight: 5 }} />
                    <Text note>{stars}</Text>
                  </View>
                  <View style={[ styles.inline, styles.ceil ]}>
                    <Ionicons name='ios-git-network-outline' style={{ marginRight: 5 }} />
                    <Text note>{forks}</Text>
                  </View>
                  <Text note>Updated at {new Date(updatedAt).toLocaleDateString()}</Text>
                </View>
              </View>
            </ListItem>
          )}
        />
      </List>
    );
  }
};

/**
 * Prop Types
 * Required user and switches to control the data source
 */
RepositoryList.propTypes = {
  user: PropTypes.string.isRequired,
  enableRepos: PropTypes.bool,
  enableStarredRepos: PropTypes.bool
};

/**
 * Get Repo List Query
 */
const GetReposQuery = gql`
  query GetRepos($login: String!, $after: String) {
    user(login: $login) {
      repositories(first: 20, after: $after) {
        edges {
          node {
            id
            name
            nameWithOwner
            description
            updatedAt
            isFork
            owner {
              login
            }
            parent {
              nameWithOwner
            }
            stargazers {
              totalCount
            }
            forks {
              totalCount
            }
            primaryLanguage {
              color
              name
            }
            licenseInfo {
              nickname
              name
            }
          }
          cursor
        }
        pageInfo {
          hasNextPage
        }
      }
    }
  }
`;

/**
 * Get starred repo query
 */
const GetStarredReposQuery = gql`
  query GetStarredRepos($login: String!, $after: String) {
    user(login: $login) {
      starredRepositories(first: 10, after: $after) {
        edges {
          node {
            id
            name
            nameWithOwner
            description
            updatedAt
            isFork
            owner {
              login
            }
            parent {
              nameWithOwner
            }
            stargazers {
              totalCount
            }
            forks {
              totalCount
            }
            primaryLanguage {
              color
              name
            }
            licenseInfo {
              nickname
              name
            }
          }
          cursor
        }
        pageInfo {
          hasNextPage
        }
      }
    }
  }
`;

/**
 * Apollo Container
 *
 * Will fetch data from server, and return a function which can fetch the next page of data and can
 * be called by the component.
 */
const withRepos = graphql(GetReposQuery, {
  name: 'repos',
  skip: ({ enableRepos }) => !enableRepos,
  options: (props) => ({
    variables: {
      login: props.user
    },
    notifyOnNetworkStatusChange: true
  }),
  props: ({ repos, ownProps }) => {
    let data = repos;

    if (data.loading) {
      return { loading: true, fetchNextPage: () => {} };
    }

    if (data.error || !data.user) {
      console.log(data.error);
    }

    const fetchNextPage = () => {
      return data.fetchMore({
        variables: {
          after: _.last(data.user.repositories.edges).cursor,
        },
        updateQuery: (previousResult, { fetchMoreResult }) => {
          return Object.assign({}, previousResult, {
            user: {
              __typename: previousResult.user.__typename,
              repositories: {
                __typename: previousResult.user.repositories.__typename,
                edges: [ ...previousResult.user.repositories.edges, ...fetchMoreResult.user.repositories.edges ],
                pageInfo: fetchMoreResult.user.repositories.pageInfo,
              }
            }
          });
        }
      });
    };

    return {
      repos: {
        repository: data.user.repositories,
        hasNextPage: data.user.repositories.pageInfo.hasNextPage,
        fetchNextPage,
        refetch: data.refetch
      }
    };
  }
});

/**
 * Apollo container
 * Same as previous container
 */
const withStarredRepos = graphql(GetStarredReposQuery, {
  name: 'starredRepos',
  skip: ({ enableStarredRepos }) => !enableStarredRepos,
  options: (props) => ({
    variables: {
      login: props.user
    },
    notifyOnNetworkStatusChange: true
  }),
  props: ({ starredRepos, ownProps }) => {
    let data = starredRepos;

    if (data.loading || !data.user) {
      return { loading: true, fetchNextPage: () => {} };
    }

    if (data.error) {
      console.log(data.error);
    }

    const fetchNextPage = () => {
      return data.fetchMore({
        variables: {
          after: _.last(data.user.starredRepositories.edges).cursor,
        },
        updateQuery: (previousResult, { fetchMoreResult }) => {
          return Object.assign({}, previousResult, {
            user: {
              __typename: previousResult.user.__typename,
              starredRepositories: {
                __typename: previousResult.user.starredRepositories.__typename,
                edges: [ ...previousResult.user.starredRepositories.edges, ...fetchMoreResult.user.starredRepositories.edges ],
                pageInfo: fetchMoreResult.user.starredRepositories.pageInfo,
              }
            }
          });
        }
      });
    };

    return {
      repos: {
        repository: data.user.starredRepositories,
        hasNextPage: data.user.starredRepositories.pageInfo.hasNextPage,
        fetchNextPage,
        refetch: data.refetch
      }
    };
  }
});

export default connect()(withStarredRepos(withRepos(RepositoryList)));