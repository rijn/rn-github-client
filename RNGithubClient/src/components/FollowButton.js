import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View, FlatList, Dimensions } from 'react-native';
import {
  Container, Content, Separator, List, ListItem, Text, Left, Body, Right,
  Thumbnail, Button, Spinner, Toast
} from 'native-base';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import {
  followUser as callFollowUser,
  unfollowUser as callUnfollowUser
} from '../services/githubFollow';

import ErrorToast from './ErrorToast';

const styles = StyleSheet.create({
});

class FollowButton extends React.Component {
  constructor(props) {
    super(props);
  }

  async followUser() {
    let { login } = this.props;
    try {
      const res = await callFollowUser(login);
      this.props.data.refetch();
    } catch (e) {
      ErrorToast(e);
    }
  }

  async unfollowUser() {
    let { login } = this.props;
    try {
      const res = await callUnfollowUser(login);
      this.props.data.refetch();
    } catch (e) {
      ErrorToast(e);
    }
  }

  render() {
    let { dispatch, data } = this.props;

    console.log(this.props);

    if (!data || data.loading) {
      return <Spinner color='#00a6de' />;
    }

    let { user: { viewerCanFollow, viewerIsFollowing } } = data;

    if (!viewerCanFollow) {
      return <View></View>
    }

    return (
      <Button small info bordered={viewerIsFollowing}
        onPress={() => { viewerIsFollowing ? this.unfollowUser() : this.followUser(); }}>
        <Text>{viewerIsFollowing ? 'Followed' : 'Follow'}</Text>
      </Button>
    );
  }
};

FollowButton.propTypes = {
  login: PropTypes.string.isRequired
};

const GetFollowStatusQuery = gql`
  query GetFollowStatus($login: String!) {
    user(login: $login) {
      viewerCanFollow
      viewerIsFollowing
    }
  }
`;

const withInfo = graphql(GetFollowStatusQuery, {
  options: (props) => ({
    variables: props
  })
});

export default connect()(withInfo(FollowButton));