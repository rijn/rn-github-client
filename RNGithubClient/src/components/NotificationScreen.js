import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View, FlatList } from 'react-native';
import { Container, Content, Separator, List, ListItem, Text, Left, Body, Right, Thumbnail } from 'native-base';
import ErrorToast from './ErrorToast';

import { getNotification } from '../services/githubNotification';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  }
});

class SettingScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      notifications: [],
      loading: false,
      refreshing: true
    };
  }

  componentWillReceiveProps ({ currentRoute }) {
    if (!currentRoute) return;
    if (currentRoute.key == 'Notification' && this.state.notifications.length == 0) {
      this.handleRequest()
    }
  }

  handleRequest = (before) => {
    this.setState({ loading: true });
    console.log(before);
    return getNotification(before).then(res => {
      this.setState({
        notifications: _.uniqBy(_.concat(this.state.notifications, res), 'id')
      });
      this.setState({ loading: false, refreshing: false });
    }).catch(e => {
      ErrorToast(e);
      this.setState({ loading: false, refreshing: false });
    });
  }

  handleRefresh = () => {
    this.setState({
      refreshing: true,
      notifications: []
    }, () => { this.handleRequest(); });
  }

  handleLoadMore = ({ distanceFromEnd }) => {
    if (this.state.loading) return;
    console.log(this.state.notifications.length);
    this.handleRequest(_.last(this.state.notifications).updated_at);
  }

  convertDate = (date) => {
    if (!_.isDate(date)) date = new Date(date);
    var seconds = Math.floor((new Date() - date) / 1000);
    var interval = Math.floor(seconds / 31536000);
    if (interval > 1) {
      return interval + " years";
    }
    interval = Math.floor(seconds / 2592000);
    if (interval > 1) {
      return interval + " months";
    }
    interval = Math.floor(seconds / 86400);
    if (interval > 1) {
      return interval + " days";
    }
    interval = Math.floor(seconds / 3600);
    if (interval > 1) {
      return interval + " hours";
    }
    interval = Math.floor(seconds / 60);
    if (interval > 1) {
      return interval + " minutes";
    }
    return Math.floor(seconds) + " seconds";
  }

  render() {
    let { dispatch } = this.props;
    let { notifications, loading } = this.state;

    return (
      <Container>
        <Content>
          <View style={styles.container}>
            <List scrollEnabled={false} style={styles.container}>
              <FlatList
                data={notifications}
                refreshing={this.state.refreshing}
                onEndReached={this.handleLoadMore}
                onEndReachedThreshold={0}
                onRefresh={this.handleRefresh}
                keyExtractor={({ id }) => id}
                renderItem={({ item: { id, repository: { full_name }, subject: { title, url }, updated_at } }) => (
                  <ListItem avatar onPress={() => { dispatch({ type: 'WebView', params: { url: url.replace('api.github.com/repos', 'github.com') } }); }}>
                    <Left>
                    </Left>
                    <Body>
                      <Text note>{full_name}</Text>
                      <Text>{title}</Text>
                    </Body>
                    <Right>
                      <Text note>{this.convertDate(updated_at)}</Text>
                    </Right>
                  </ListItem>
                )}
              />
            </List>
          </View>
        </Content>
      </Container>
    );
  }
};

const mapStateToProps = ({ nav }) => {
  if (!nav.routes[nav.index].routes) return;
  const currentRouteIndex = nav.routes[nav.index].index;
  const currentRoute = nav.routes[nav.index].routes[currentRouteIndex];

  return { currentRoute }
}

export default connect(mapStateToProps)(SettingScreen);