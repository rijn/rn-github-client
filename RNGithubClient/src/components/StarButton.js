import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { StyleSheet, View, FlatList, Dimensions, Button } from 'react-native';
import {
  Container, Content, Separator, List, ListItem, Text, Left, Body, Right,
  Thumbnail, Spinner, Toast
} from 'native-base';

import Ionicons from 'react-native-vector-icons/Ionicons';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import {
  starRepo,
  unstarRepo
} from '../services/githubStar';

import ErrorToast from './ErrorToast';

const styles = StyleSheet.create({
});

export class StarButton extends React.Component {
  constructor(props) {
    super(props);
  }

  async changeStarRepo(viewerHasStarred) {
    try {
      const res = viewerHasStarred ? await unstarRepo(this.props) : await starRepo(this.props);
      this.props.data.refetch();
    } catch (e) {
      ErrorToast(e);
    }
  }

  render() {
    let { dispatch, data } = this.props;

    if (!data || data.loading) {
      return <Spinner color='#00a6de' />;
    }

    let { repository: { viewerHasStarred } } = data;

    return (
      <Ionicons
        name={viewerHasStarred ? 'ios-star' : 'ios-star-outline'}
        size={20} color='#00a6de'
        onPress={() => { this.changeStarRepo(viewerHasStarred); }} />
    );
  }
};

StarButton.propTypes = {
  owner: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};

const GetStarStatusQuery = gql`
  query GetFollowStatus($owner: String!, $name: String!) {
    repository(owner: $owner, name: $name) {
      viewerHasStarred
    }
  }
`;

const withInfo = graphql(GetStarStatusQuery, {
  options: (props) => ({
    variables: props
  })
});

export default connect()(withInfo(StarButton));