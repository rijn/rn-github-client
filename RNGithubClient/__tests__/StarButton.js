import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import gql from 'graphql-tag';
import { mockServer, makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { graphql, ApolloProvider } from 'react-apollo';
import { ApolloClient, createNetworkInterface } from 'apollo-client';
import { mockNetworkInterfaceWithSchema } from 'apollo-test-utils';
import { StarButton } from '../src/components/StarButton';

import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store'

test('renders loading correctly', () => {
  const tree = renderer.create(
    <StarButton owner='test' name='test' data={{
      loading: true,
    }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders has not starred correctly', () => {
  let tree = renderer.create(
    <StarButton owner='test' name='test' data={{
      loading: false,
      repository: {
        viewerHasStarred: false
      }
    }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders has starred correctly', () => {
  let tree = renderer.create(
    <StarButton owner='test' name='test' data={{
      loading: false,
      repository: {
        viewerHasStarred: true
      }
    }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});