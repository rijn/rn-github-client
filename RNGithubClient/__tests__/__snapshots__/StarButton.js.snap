// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders has not starred correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#00a6de",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders has starred correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#00a6de",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders loading correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#00a6de"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "height": 80,
    }
  }
  virtual={undefined}
/>
`;
